internal/cluster/cluster_test.go

```diff
@@ -1156,6 +1270,12 @@ var _ = Describe("Auto assign machine CIDR", func() {
 				Expect(network.GetMachineCidrById(&cluster, 0)).To(Equal(t.expectedMachineCIDR))
 				Expect(cluster.MachineNetworkCidr).To(Equal("")) // TODO(MGMT-9751-remove-single-network)
 			}
+			if t.expectedMachineNetworks != nil {
+				Expect(cluster.MachineNetworks).To(HaveLen(len(t.expectedMachineNetworks)))
+				for _, m := range t.expectedMachineNetworks {
+					Expect(t.expectedMachineNetworks).To(ContainElement(m))
+				}
+			}
```

Comment:
Thanks a lot for this part! I should have added it in another PR, but I see you have it covered